### PR TITLE
✨Adding ConsentManager CMP to amp-consent

### PR DESCRIPTION
**Pull request by janwinkler**
Added ConsentManager CMP (www.consentmanager.net) to amp-consent for easier GDPR consent support.

### CODE REVIEW

**Comment by janwinkler** [comment]
@googlebot I fixed it.

**Comment by janwinkler** [comment]
@googlebot I fixed it.

**Comment by janwinkler** [comment]
@googlebot I consent.

**Comment by carsten1980** [comment]
@googlebot I consent.

File: examples/amp-consent/cmp-vendors.amp.html [hunk 1]
```diff
@@ -96,6 +96,7 @@
           <!-- This should be in alphabetical order. _ping_ should be the first -->
           <option>_ping_</option>
           <option>appconsent</option>
+          <option>ConsentManager</option>
```
Comment (micajuine-ho) [inline examples/amp-consent/cmp-vendors.amp.html:99]: nit: move above Marfeel

File: examples/amp-consent/cmp-vendors.amp.html [hunk 2]
```diff
@@ -159,6 +160,22 @@ <h3>Image that is NOT blocked by consent</h3>
       </amp-consent>
       <!-- End appconsent example -->
 
+
+      <!-- ConsentManager example -->
+      <amp-consent id="ConsentManager" layout="nodisplay" type="ConsentManager">
```
Comment (micajuine-ho) [inline examples/amp-consent/cmp-vendors.amp.html:165]: nit: move this example above Marfeel

File: extensions/amp-consent/cmps/consentmanager.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<!---
+Copyright 2019 The AMP HTML Authors. All Rights Reserved.
```
Comment (micajuine-ho) [inline extensions/amp-consent/cmps/consentmanager.md:2]: nit: 2020

**Comment by janwinkler** [comment]
> Additionally, is there a test ID that I could use to ensure that the `checkConsentHref` and `promptUiSrc` endpoints are working correctly in the implementation?

Please try id 10640

**Comment by micajuine-ho** [comment]
> Please try id 10640

The `checkConsentHref` is working; but I am getting the following response from the `promptUiSrc` endpoint:
`ID not found`

**Comment by janwinkler** [comment]
> 
> 
> > Please try id 10640
> 
> The `checkConsentHref` is working; but I am getting the following response from the `promptUiSrc` endpoint:
> `ID not found`

Can you send the browser request (URL + Post Body) that is sent to our server? we assumed that promptUiSrc will receive the same data as checkConsentHref via HTTP POST!?

**Comment by micajuine-ho** [comment]
> we assumed that promptUiSrc will receive the same data as checkConsentHref via HTTP POST!?

The promptUiSrc does not receive any data in the request. Instead, [information is passed to the iframe upon creation via the window.name property](https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/integrating-consent.md#client-information-passed-to-iframe).

**Comment by janwinkler** [comment]
@micajuine-ho Can you try again please?

File: examples/amp-consent/cmp-vendors.amp.html [hunk 4]
```diff
@@ -159,6 +160,22 @@ <h3>Image that is NOT blocked by consent</h3>
...
+        <script type="application/json">
+          {
+            "postPromptUI": "postPromptUI",
+            "clientConfig": { "id": "1" }
+          }
+        </script>
+        <div id="postPromptUI">
+          Post Prompt UI
+          <button on="tap:consent.prompt(consent=ConsentManager)" role="button">Manage</button>
```
Comment (micajuine-ho) [inline examples/amp-consent/cmp-vendors.amp.html:174]: on="tap:ConsentManager.prompt(consent=ConsentManager)"

**Comment by janwinkler** [comment]
do i need to do anything from our end at this stage in order to move forward?

**Comment by micajuine-ho** [comment]
Still having some issues with prettier. 

image

**Review by janwinkler — COMMENTED** [review]
please approve

**Comment by micajuine-ho** [comment]
> not 100% sure what to do. can you elaborate?

In cmps.js, add a new line on line 47.

**Comment by janwinkler** [comment]
sorry i think im blind ;-) 
i dont see where to put it. the lines look exactly like the lines from the other cmps to me.

**Comment by micajuine-ho** [comment]
Line 47 goes past our character limit, so need to ad a new line after the colon:

```
'checkConsentHref':
    'https://consentmanager.mgr.consensu.org/delivery/ampcheck.php',
```

Make sure no extra white space after the colon and the correct tab index before the link.

**Comment by janwinkler** [comment]
done :-)

**Review by janwinkler — COMMENTED** [review]
Fixed

**Review by micajuine-ho — APPROVED** [review]

**Comment by janwinkler** [comment]
seems everything is fine now? when can we expect that it will be available for public use? what's the timing for that?

**Comment by micajuine-ho** [comment]
2 weeks for it to be in prod.

**Pull request merged**